manifests/init.pp

```diff
@@ -438,6 +480,123 @@
       enable  => $enable_pcsd_service,
       require => Package['pcs'],
     }
+
+    # Determine if this node should perform authorizations
+    case $manage_pcsd_auth_node {
+      'first': { $auth_node = $quorum_members[0] }
+      'last': { $auth_node = $quorum_members[length($quorum_members)-1] }
+      default: {}
+    }
+
+    # If the local data matches auth_node (hostname or primary IP) we can
+    # perform auth processing for subsequent components
+    if $trusted['certname'] == $auth_node or
+        $facts['networking']['ip'] == $auth_node {
+          $is_auth_node = true
+    } else {
+      $is_auth_node = false
+    }
+
+    $exec_path = '/sbin:/bin/:usr/sbin:/usr/bin'
+
+    if $manage_pcsd_auth and $is_auth_node {
+      if ! $sensitive_hacluster_password or ! $sensitive_hacluster_hash {
+        fail('The hacluster password and hash must be provided to authorize nodes via pcsd.')
+      }
+
+      # TODO - verify if this breaks out of the sensitivity
+      $hacluster_password = $sensitive_hacluster_password.unwrap
+      $auth_credential_string = "-u hacluster -p ${hacluster_password}"
+
+      # As the auth can happen before corosync.conf exists we need to explicitly
+      # list the members to join. 
+      # TODO - verify that this is safe when quorum_members is a list of IP
+      # addresses
+      $node_string = join($quorum_members, ' ')
```

Comment:
I do have standard rspec tests which verify that the config is generated as expected, however, I don't know whether it would work with PCS in practice. At this point I haven't worked with beaker / acceptance testing so my "acceptance" tests are via application to my actual environment and I don't use IP address based clusters.